src/Frontend/Core/Header/GoogleAnalytics.php

```diff
@@ -4,48 +4,49 @@
 
 use Common\Core\Cookie;
 use Common\ModulesSettings;
+use ForkCMS\Privacy\ConsentDialog;
 
 final class GoogleAnalytics
 {
     /** @var ModulesSettings */
     private $modulesSettings;
 
-    /** @var string */
-    private $httpHost;
+    /**
+     * @var ConsentDialog
+     */
+    private $consentDialog;
 
-    /** @var Cookie */
-    private $cookie;
-
-    public function __construct(ModulesSettings $modulesSettings, string $httpHost, Cookie $cookie)
+    public function __construct(ModulesSettings $modulesSettings, ConsentDialog $consentDialog)
     {
         $this->modulesSettings = $modulesSettings;
-        $this->httpHost = $httpHost;
-        $this->cookie = $cookie;
+        $this->consentDialog = $consentDialog;
     }
 
     private function shouldAddGoogleAnalyticsHtml(): bool
     {
-        $siteHTMLHeader = (string) $this->modulesSettings->get('Core', 'site_html_header', '');
-        $siteHTMLFooter = (string) $this->modulesSettings->get('Core', 'site_html_footer', '');
-        $webPropertyId = (string) $this->modulesSettings->get('Analytics', 'web_property_id', null);
+        $googleAnalyticsTrackingId = $this->modulesSettings->get(
+            'Core',
+            'google_tracking_google_analytics_tracking_id',
+            ''
+        );
 
-        return $webPropertyId !== ''
-               && mb_strpos($siteHTMLHeader, $webPropertyId) === false
-               && mb_strpos($siteHTMLFooter, $webPropertyId) === false;
+        return ($googleAnalyticsTrackingId !== '');
     }
 
     private function shouldAnonymize(): bool
     {
-        return $this->modulesSettings->get('Core', 'show_cookie_bar', false) && !$this->cookie->hasAllowedCookies();
-    }
+        // if the consent dialog is disabled we will anonymize by default
+        if (!$this->modulesSettings->get('Core', 'show_consent_dialog', false)) {
+            return true;
```

Comment:
I would have it fallback here on the cookiebar and mark it as deprecated

Reply:
Fixed in c96374567c5503b16ae091b4121883be50ce9d41